internal/wasm/jit/jit_impl_test.go

```diff
@@ -0,0 +1,4448 @@
+				if shouldGoElse {
+					x2, x1 = x1, x2
+				}
+				requirePushTwoFloat32Consts(t, x1, x2, compiler)
+				// Lt on floats produces the value on COND_MI register.
+				err := compiler.compileLt(&wazeroir.OperationLt{Type: wazeroir.SignedTypeFloat32})
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "EQ",
+			setupFunc: func(t *testing.T, compiler compilerImpl, shouldGoElse bool) {
+				x1, x2 := uint32(1), uint32(1)
+				if shouldGoElse {
+					x2++
+				}
+				requirePushTwoInt32Consts(t, x1, x2, compiler)
+				err := compiler.compileEq(&wazeroir.OperationEq{Type: wazeroir.UnsignedTypeI32})
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "NE",
+			setupFunc: func(t *testing.T, compiler compilerImpl, shouldGoElse bool) {
+				x1, x2 := uint32(1), uint32(2)
+				if shouldGoElse {
+					x2 = x1
+				}
+				requirePushTwoInt32Consts(t, x1, x2, compiler)
+				err := compiler.compileNe(&wazeroir.OperationNe{Type: wazeroir.UnsignedTypeI32})
+				require.NoError(t, err)
+			},
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			for _, shouldGoToElse := range []bool{false, true} {
+				shouldGoToElse := shouldGoToElse
+				t.Run(fmt.Sprintf("should_goto_else=%v", shouldGoToElse), func(t *testing.T) {
+					env := newJITEnvironment()
+					compiler := env.requireNewCompiler(t, nil)
+					err := compiler.compilePreamble()
+					require.NoError(t, err)
+
+					tc.setupFunc(t, compiler, shouldGoToElse)
+					require.Equal(t, uint64(1), compiler.valueLocationStack().sp)
+
+					err = compiler.compileBrIf(&wazeroir.OperationBrIf{Then: thenBranchTarget, Else: elseBranchTarget})
+					require.NoError(t, err)
+					compiler.compileExitFromNativeCode(unreachableStatus)
+
+					// Emit code for .then label.
+					skip := compiler.compileLabel(&wazeroir.OperationLabel{Label: thenBranchTarget.Target.Label})
+					require.False(t, skip)
+					compiler.compileExitFromNativeCode(thenLabelExitStatus)
+
+					// Emit code for .else label.
+					skip = compiler.compileLabel(&wazeroir.OperationLabel{Label: elseBranchTarget.Target.Label})
+					require.False(t, skip)
+					compiler.compileExitFromNativeCode(elseLabelExitStatus)
+
+					code, _, _, err := compiler.compile()
+					require.NoError(t, err)
+
+					// The generated code looks like this:
+					//
+					//    ... code from compilePreamble()
+					//    ... code from tc.setupFunc()
+					//    br_if .then, .else
+					//    exit $unreachableStatus
+					// .then:
+					//    exit $thenLabelExitStatus
+					// .else:
+					//    exit $elseLabelExitStatus
+					//
+					// Therefore, if we start executing from the top, we must end up exiting with an appropriate status.
+					env.exec(code)
+					require.NotEqual(t, unreachableStatus, env.jitStatus())
+					if shouldGoToElse {
+						require.Equal(t, elseLabelExitStatus, env.jitStatus())
+					} else {
+						require.Equal(t, thenLabelExitStatus, env.jitStatus())
+					}
+				})
+			}
+		})
+	}
+}
```

Comment:
this this file is too large. can you split it into files so that each are (ideally)_ not more than 500 lines? I suspect that at least >1.0 feature can be separate file, and then the 1.0 split by category to achieve this.